rpc/rpcevents/execution_events_server.go

```diff
@@ -193,14 +193,15 @@ func (ees *executionEventsServer) subscribeBlockExecution(ctx context.Context, c
 	return nil
 }
 
-func (ees *executionEventsServer) iterateBlockEvents(start, end uint64, consumer func(*exec.StreamEvent) error) (lastHeightSeen uint64, err error) {
-	err = ees.eventsProvider.IterateStreamEvents(start, end, func(blockEvent *exec.StreamEvent) error {
-		if blockEvent.TxExecution == nil {
-			// nil safe
-			lastHeightSeen = blockEvent.GetEndBlock().GetHeight()
+func (ees *executionEventsServer) iterateBlockEvents(start, end uint64, consumer func(*exec.StreamEvent) error) (uint64, error) {
+	// Assume that we have seen the previous block before start to have ended up here
+	lastHeightSeen := start - 1
```

Comment:
That looks like a an underflow waiting to happen, i.e. if we iterate from block 0 then lastHeightSeen will be -1 and the leveldb reader does not deal with negatives well.

Reply:
You are right, I have added a comment that notes this is true but we get away with it in this case.